workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/BackStackContainer.kt

```diff
@@ -0,0 +1,233 @@
+import androidx.savedstate.ViewTreeSavedStateRegistryOwner
+import androidx.transition.Scene
+import androidx.transition.Slide
+import androidx.transition.TransitionManager
+import androidx.transition.TransitionSet
+import com.squareup.workflow1.ui.NamedScreen
+import com.squareup.workflow1.ui.R
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.androidx.WorkflowAndroidXSupport.stateRegistryOwnerFromViewTreeOrContext
+import com.squareup.workflow1.ui.androidx.WorkflowLifecycleOwner
+import com.squareup.workflow1.ui.buildView
+import com.squareup.workflow1.ui.canShowRendering
+import com.squareup.workflow1.ui.compatible
+import com.squareup.workflow1.ui.container.BackStackConfig.First
+import com.squareup.workflow1.ui.container.BackStackConfig.Other
+import com.squareup.workflow1.ui.container.ViewStateCache.SavedState
+import com.squareup.workflow1.ui.showFirstRendering
+import com.squareup.workflow1.ui.showRendering
+
+/**
+ * A container view that can display a stream of [BackStackScreen] instances.
+ *
+ * This container supports saving and restoring the view state of each of its subviews corresponding
+ * to the renderings in its [BackStackScreen]. It supports two distinct state mechanisms:
+ *  1. Classic view hierarchy state ([View.onSaveInstanceState]/[View.onRestoreInstanceState])
+ *  2. AndroidX [SavedStateRegistry] via [ViewTreeSavedStateRegistryOwner].
+ *
+ * ## A note about `SavedStateRegistry` support.
+ *
+ * The [SavedStateRegistry] API involves defining string keys to associate with state bundles. These
+ * keys must be unique relative to the instance of the registry they are saved in. To support this
+ * requirement, [BackStackContainer] tries to generate a best-effort unique key by combining its
+ * fully-qualified class name with both its [view ID][View.getId] and the
+ * [compatibility key][com.squareup.workflow1.ui.Compatible.compatibilityKey] of its rendering.
+ *
+ * This method isn't guaranteed to give a unique registry key, but it should be good enough: If you
+ * need to nest multiple [BackStackContainer]s under the same `SavedStateRegistry`, just wrap each
+ * [BackStackScreen] with a [NamedScreen], or give each [BackStackContainer] a unique view ID. If that
+ * heuristic fails you, use [ViewEnvironment.withBackStackStateKeyPrefix] to add unique names to
+ * the [ViewEnvironment] used to show each [BackStackScreen].
+ *
+ * There's a potential issue here where if our ID is changed to something else, then another
+ * [BackStackContainer] is added with our old ID, that container will overwrite our state. Since
+ * they'd both be using the same key, [SavedStateRegistry] would throw an exception. As long as this
+ * container is detached before its ID is changed, it shouldn't be a problem.
+ */
+@WorkflowUiExperimentalApi
+public open class BackStackContainer @JvmOverloads constructor(
```

Comment:
Almost nothing changed here, beyond moving the class and replacing `Named` with `NamedScreen`. I don't know why github won't show a diff.